### DISCUSSION TITLE
Why is initEmail commented out?

### DISCUSSION THREAD

**Discussion by toddheslin** [nocodb/nocodb#279] [Discord Threads]
This isn't as much of a bug as a question. We set up an instance and the product owner set up their account and they added me to the instance. I didn't receive an email so I checked the logs and found:

```
xc_1    | Warning : `mailSend` failed, Please configure emailClient configuration.
xc_1    | Password reset token : xxxxxxxxxxxxxxxxxxxxxx
xc_1    | POST /auth/password/forgot 200 51 - 16.826 ms
```
Which I figured out I could just hit the reset endpoint with this token (`/password/reset/xxxxxxxxxxxxxxxxxxxxxx`)and I'd be good to go. That worked.

But when I was looking how to configure the email adapter it looks like it has been commented out in the source code. I'm just a bit confused why this is.

https://github.com/nocodb/nocodb/blob/master/packages/nocodb/src/lib/noco/meta/NcMetaMgr.ts#L979-L988

Thanks!

**Comment by pranavxc** [comment]
That's our older implementation and we are no longer using it. Currently, we are initializing it from `NcPluginMgr` and you have to provide SMTP details under AppStore.

**Reply by o1lab** [reply]
Moved this to discussion.

**Reply by Cherylyin** [reply]
> That's our older implementation and we are no longer using it. Currently, we are initializing it from `NcPluginMgr` and you have to provide SMTP details under AppStore.

test

**Reply by Cherylyin** [reply]
> That's our older implementation and we are no longer using it. Currently

test

**Comment by a5r0n** [comment]
Just to point out for anyone looking for how to reset the password when an email is not sent,
`/password/reset/:token`

**Comment by gramakri** [comment]
Is it possible to configure the SMTP details via code? It seems that `NC_MAILER` and related env are not read as part of the default NcPluginMgr initialization.

**Reply by gramakri** [reply]
What I mean is https://github.com/nocodb/nocodb/blob/f0bab968324ab9d737952defc142f278d9d4fe88/packages/nocodb/src/lib/utils/NcConfigFactory.ts#L385 is read in only for makeProjectConfigFromUrl() but not in make() . But as I understand, we want to use the Appstore Email configuration anyways.

So, is it possible to initialize SMTP configurable variables using code as part of https://github.com/nocodb/nocodb-seed/blob/main/index.js ?

**Comment by o1lab** [comment]
MAILER related env variables are not being used. The idea with this is we want to make it mostly nocode so that it could be done via GUI.

**Reply by gramakri** [reply]
Atleast from Cloudron point of view, we consider things like configuring email and databases as "provisioning"/"sysadmin" stuff, as opposed to a coding/developer responsibility. Would you agree with this? (This is true for other apps, but I don't know about nocodb enough yet to understand this better). Can the email delivery be configured using code inside nocodb?

**Reply by o1lab** [reply]
> Can the email delivery be configured using code inside nocodb?

Yes, via app store (bottom left within project).

However, we will think about making mailer related things via env variables.
